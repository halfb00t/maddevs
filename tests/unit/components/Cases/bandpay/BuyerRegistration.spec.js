import { render, screen } from '@testing-library/vue'
import { createLocalVue, mount } from '@vue/test-utils'
import BuyerRegistration from '@/components/Cases/bandpay/BuyerRegistration'
import '../../../__mocks__/intersectionObserverMock'
// eslint-disable-next-line import/order
import lazyLoad from 'nuxt-lazy-load/lib/module'

const mocks = {
  $getMediaFromS3: () => 'img.jpg',
}

const directives = {
  'lazy-load': () => {},
}

const localVue = createLocalVue()
localVue.directive('lazy-load', lazyLoad)
jest.mock('nuxt-lazy-load/lib/module')

describe('BandPay BuyerRegistration component', () => {
  it('should render correctly', () => {
    const { container } = render(BuyerRegistration, {
      mocks,
      directives,
      localVue,
    })

    expect(container).toMatchSnapshot()
  })

  it('should render with text', () => {
    render(BuyerRegistration, {
      mocks,
      directives,
      localVue,
    })

    expect(screen.getByText(/Buyer registration/i).className).toBeTruthy()
  })

  it('should render paragraphs', () => {
    const wrapper = mount(BuyerRegistration, {
      mocks,
      directives,
      localVue,
    })

    const paragraphs = wrapper.findAll('.case_paragraph')
    expect(paragraphs).toHaveLength(3)
  })

  it('should render image', () => {
    render(BuyerRegistration, {
      mocks,
      directives,
      localVue,
    })

    expect(screen.getByTestId('test-picture-img').getAttribute('data-src'))
      .toBe(mocks.$getMediaFromS3())
    expect(screen.getByTestId('test-picture-img').getAttribute('width')).toBe('1028')
    expect(screen.getByTestId('test-picture-img').getAttribute('height')).toBe('612')
  })
})
