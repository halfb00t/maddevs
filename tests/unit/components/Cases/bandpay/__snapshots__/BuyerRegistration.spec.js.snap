// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BandPay BuyerRegistration component should render correctly 1`] = `
<div>
  <section
    class="m-80_bottom media-m-48_bottom"
  >
    <section
      class="container_regular m-24_bottom"
    >
      <h3
        class="case_title_h2 m-34_bottom"
      >
        
      Buyer registration
    
      </h3>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      Registration with BandPay takes place only in the mobile application. However,
      the buyer and the seller can start cooperating via a webform, too.
    
      </p>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      For example, the buyer can go to the seller’s public profile and create a project request.
      With that, the system will ask the buyer to add contact information. The confirmation link will
      then be sent to the buyer’s email address for verification. The buyer then becomes a registered user.
      After that, the seller sees the request and creates an agreement.
    
      </p>
       
      <p
        class="case_paragraph m-24_bottom media-m-12_bottom"
      >
        
      Alternatively, the seller can create an instant deposit in the app and notify the buyer via email.
      At the same time, BandPay will send an invoice to the buyer’s email address.
      After paying the invoice, the buyer becomes a registered user.
    
      </p>
    </section>
     
    <section
      class="container_middle"
    >
      <div
        class="case_image-wrapper"
      >
        <picture>
          <source
            class="image"
            data-srcset="img.jpg ,img.jpg"
            type="image/webp"
          />
           
          <img
            alt="Buyer Registration"
            class="image"
            data-src="img.jpg"
            data-srcset="img.jpg"
            data-testid="test-picture-img"
            height="612"
            width="1028"
          />
        </picture>
      </div>
    </section>
  </section>
</div>
`;
